src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -10,37 +11,70 @@ import {
   ModalHeader,
   useColorModeValue,
 } from '@chakra-ui/react'
+import { Vault } from '@shapeshiftoss/hdwallet-native-vault'
+import * as bip39 from 'bip39'
 import { useState } from 'react'
 import { FieldValues, useForm } from 'react-hook-form'
 import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router-dom'
 import { Card } from 'components/Card/Card'
 import { Text } from 'components/Text'
 
+import { LoginError } from '../types'
+
 // @TODO(NeOMakinG): Remove this and add the 2fa code logic
 const twoFactorAuthCode = ''
 
+// @TODO(NeOMakinG): Change this with the mnemonic of the legacy account
+const DUMMY_MNEMONIC =
+  'yawning yodellers yielded yearningly yet yodeling yeti yelped yearningly yet youths yawned'
+
 export const LegacyLogin = () => {
+  const history = useHistory()
   const [isCaptchaSolved, setIsCaptchaSolved] = useState(false)
   const [error, setError] = useState<boolean | string>(false)
+  const [isTwoFactorRequired, setTwoFactorRequired] = useState(false)
   const captchaBgColor = useColorModeValue('gray.50', 'gray.700')
   const checkboxBorderColor = useColorModeValue('gray.400', 'white')
 
   const {
     handleSubmit,
     register,
-    reset,
     formState: { errors, isSubmitting },
   } = useForm({ shouldUnregister: true })
 
   const translate = useTranslate()
 
+  const isLoginError = (err: any): err is LoginError => err && typeof err.message === 'string'
+
   const onSubmit = async (values: FieldValues) => {
-    // @TODO(NeOMakinG): Remove this when we handle the error response from the API
-    if (true) {
+    try {
+      // @TODO: Replace this with API call when backend is ready
+      const { mnemonic } = await Promise.resolve({ mnemonic: DUMMY_MNEMONIC })
+
+      if (!bip39.validateMnemonic(mnemonic)) {
+        // @TODO: Should we send an error to the user?
+      }
+
+      // TODO: use response to create wallet vault.
+      const vault = await Vault.create(undefined, false)
+      vault.meta.set('createdAt', Date.now())
+      vault.set('#mnemonic', mnemonic)
+      history.push('/native/legacy/login/success', { vault })
+    } catch (err) {
+      if (isLoginError(err) && err.message === '2fa required') {
```

Comment:
see string union comments

Reply:
It might be better to do this in the upcoming PR to wire up the calls. We don't have the backend functionality for 2fa yet. I don't want to ossify the front end for that error that may change.